tests/examples/langchain/medical_rag/diagnosis_agent.py

```
@@ -0,0 +1,159 @@
+        PROMPT =  PromptTemplate(
+            template=prompt_template,
+            input_variables=["context", "disease", "history", "symptoms"]
+        )
+
+        results = []
+        for candidate in matched[:3]:
+            retriever = self.vector_db.as_retriever(search_kwargs={"k": 3})
+            qa_chain = (
+                {
+                    'context': retriever, 
+                    'disease': lambda _: candidate["disease"],
```

Ruff (0.8.2)

Function definition does not bind loop variable `candidate`

(B023)